### ISSUE TITLE
ejabberd 17.08 mod_register with captcha protection is not working with proper configured settings

### ISSUE THREAD

**Issue by cmeng-git** [processone/ejabberd#1928]
> What version of ejabberd are you using?
17.08

> What operating system (version) are you using?
Ubuntu 16.04

> How did you install ejabberd (source, package, distribution)?
source built

> What did not work as expected? Are there error messages in the log? What
Below is an extraction of captcha configuration setting in ejabberd.yml. But ejabberd throw the following exception when an xmpp client is trying to perform IBR. The IBR was tested working in an earlier ejabberd version.

16:06:55.029 [error] Hook c2s_unauthenticated_packet crashed when running mod_register:c2s_unauthenticated_packet/2:
** Reason = {throw,<<"The port number mentioned in captcha_host is not a ejabberd_http listener with 'captcha' option. Change the port number or specify http:// in that option."> 

** ejabberd.yml captcha configuration **
```
listen:
  - 
    port: 5280
    ip: "::"
    module: ejabberd_http
    request_handlers:
      "/ws": ejabberd_http_ws
      "/bosh": mod_bosh
      "/api": mod_http_api
    web_admin: true
    register: true
    captcha: true

captcha_cmd: "/usr/lib/ejabberd/priv/bin/captcha.sh"
captcha_host: "example.com:5280"

modules:
  mod_register: 
    captcha_protected: true
    ip_access: all
```

> was the unexpected behavior? What was the expected result?
The similar setting was verified working with IBR in earlier ejabberd earlier version 17.04.

I have checked the configuration setting against 
https://docs.ejabberd.im/admin/configuration/#mod-register

Any help is much appreciated

**Comment by zinid** [comment]
Confirmed

**Comment by zinid** [comment]
This will be included in 17.08.
For those who face this bug and cannot upgrade, define HTTP scheme in `captcha_host` as a workaround, like:
```yaml
captcha_host: "http://example.com:5280"
```
